src/Savegame/BattleUnit.h

```diff
@@ -207,6 +207,12 @@ class BattleUnit
 	Position getLastPosition() const;
 	/// Gets the unit's position of center in voxels.
 	Position getPositionVexels() const;
+	/// Gets the unit's X position.
+	int getPositionX() const;
```

Comment:
One detail, could you not add this functions to "public" API? They are needed only by scripts, usually in case like this I add local helper function like `getPositionXScript` in `cpp` file to be only visible by scripts. I already add some function like that.